controllers/resources/manager.go

```diff
@@ -79,22 +90,54 @@ func (m *manager) GenerateRemediationCR(node *corev1.Node, nhc *remediationv1alp
 	remediationCR.SetSelfLink("")
 	remediationCR.SetCreationTimestamp(metav1.Now())
 
+	owners := make([]metav1.OwnerReference, 0)
 	if nhc != nil {
-		remediationCR.SetOwnerReferences([]metav1.OwnerReference{
-			{
-				APIVersion:         nhc.APIVersion,
-				Kind:               nhc.Kind,
-				Name:               nhc.Name,
-				UID:                nhc.UID,
-				Controller:         pointer.Bool(false),
-				BlockOwnerDeletion: nil,
-			},
+		owners = append(owners, metav1.OwnerReference{
+			APIVersion:         nhc.APIVersion,
+			Kind:               nhc.Kind,
+			Name:               nhc.Name,
+			UID:                nhc.UID,
+			Controller:         pointer.Bool(false),
+			BlockOwnerDeletion: nil,
 		})
 		remediationCR.SetLabels(map[string]string{
 			"app.kubernetes.io/part-of": "node-healthcheck-controller",
 		})
 	}
 
+	// TODO also handle CAPI clusters / machines
+	if m.onOpenshift {
+		machineRef, machineNamespace, err := m.getOwningMachineWithNamespace(node)
+		if err != nil {
+			return nil, err
+		}
+		if machineRef != nil && machineNamespace != "" {
+			// Owners must be cluster scoped, or in the same namespace as their dependent
+			// Machines are always namespaced
+			if remediationCR.GetNamespace() == machineNamespace {
+				owners = append(owners, *machineRef)
+			} else {
+				// What to do if namespaces don't match?
+				// So far this is a known issue for Metal3 remediation only, and that case was checked already
+				// in the Reconciler. So just log it, but do not fail remediation.
+				m.log.Info("Not setting remediation CR's owner ref to the machine, because namespaces don't match",
+					"template ns", remediationCR.GetNamespace(),
+					"machine ns", machineNamespace)
+			}
+		}
+	}
+
```

Comment:
Nit: I think it would be nice to extract this block logic  to a method

Reply:
agree, will do in a follow up